src/gui/editorwidgets/qgsdatetimeedit.cpp

```diff
@@ -192,8 +192,12 @@ void QgsDateTimeEdit::showEvent( QShowEvent *event )
 
 void QgsDateTimeEdit::changed( const QDateTime &dateTime )
 {
-  mIsEmpty = false;
   bool isNull = dateTime.isNull();
+
+  if ( mIsNull == isNull && QgsDateTimeEdit::dateTime() == dateTime )
```

Comment:
@elpaso this is more or less along the lines of https://github.com/qgis/QGIS/pull/30100 but on a different level.

1. does it look good to you?
2. should we remove the (now redundant i think) [outer](https://github.com/qgis/QGIS/pull/30100/files#diff-570a1ee0ae596d03e7fda06c5f47eafeR66) [checks](https://github.com/qgis/QGIS/pull/30100/files#diff-570a1ee0ae596d03e7fda06c5f47eafeR260)?

Reply:
I don't remember all the details of my PR but looking at the code I think you can remove the outer checks.

Reply:
... and yes: LGTM